serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc

```diff
@@ -24,90 +22,93 @@ kubernetes:<group>/<version>/<kind>/<namespace>/<resourceName>?<attributeName>=<
-to specific attributes.
+Query parameters in URI::
++
+--
+The query parameters are also known as query string. The query parameters are defined the similar way with URLs to assign value to specific attributes.
 
-Two query parameters are available to help the engine to be more precise when querying for the given Kubernetes Resource:
+The following query parameters help the engine to be more precise when querying for a given Kubernetes resource:
 
-* **Custom Labels**: Useful to filter services in case there are more than one Service with the same `Label Selector`
-but exposing different ports. In this case we can instruct the engine that, when more than 1 service is found, it should use the one
-that have the provided label. The label is defined with the following expression, for multiple labels use `semicolon`:
-** `labels=label-name=namevalue;another-label=another-value` +
-    Example:
+* *Custom labels*: The custom labels are used to filter services in case there are more than one service with the same label selector but exposing different ports. In this case, you can instruct the engine that if more than one service is found, then the engine must use the service containing the provided label. 
++
+The label is defined with the following expression and in case of multiple labels, you can use semicolon (;):
++
+`labels=label-name=namevalue;another-label=another-value`
 +
+.Example label definition in URI
 [source,shell]
 ----
 kubernetes:v1/pod/<namespace>/<pod-name>?labels=label-name=test-label
 ----
-*** With the URI above we will instruct that, if there are more than 1 service exposing the given pod, the label
-`label-name=test-label` will be used to filter the service. If the label does not exist, the first found service will be used.
++
+Using the previous URI example, if there are more than one service exposing the given pod, the `label-name=test-label` label is used to filter the service. If the label does not exist, the first found service is used.
+
+* *Custom port name*: The custom port name is used to determine which port to use when multiple ports are configured in the target service or container. You can configure the port name to be queried using the following pattern:
++
+`port-name=<PORT_NAME>`
+--
 
-* **Custom port name**: Useful to determine which port to use when there are many ports configured in the target Service or Container.
-It can be configured using the following patter:
-** `port-name=<PORT_NAME>`: the port name to be queried.
+Based on the resource to be discovered, the Kubernetes service discovery follows specific paths as shown in the following figure:
 
+image::cloud/sw-discovery-flow.jpg[]
 
-== Configuration
+There is no specific configuration required for the Kubernetes service discovery except by using the expected URI pattern. However, the `okhttp` communication interceptor, which logs the communication between the application and the Kubernetes API is disabled by default.
```

Comment:
also, these following sentences are a little bit disconnected without a tittle, maybe keep the Configuration subtitle and group it and the dependency exclusion?

Reply:
Added a section for configuration.